$OpenBSD: patch-lib_pypy__tkinter_tklib_py,v 1.1 2013/09/19 08:39:31 sthen Exp $

Patch in OpenBSD paths. Applied upstream, albeit without ports vars:
https://bugs.pypy.org/issue1531

Also allow tkinter to use a single threaded TCL.
From upstream a2cfb94ccff5

--- lib_pypy/_tkinter/tklib.py.orig	Thu Aug  1 10:57:52 2013
+++ lib_pypy/_tkinter/tklib.py	Sun Sep 15 20:22:30 2013
@@ -1,6 +1,7 @@
 # C bindings with libtcl and libtk.
 
 from cffi import FFI
+import sys
 
 tkffi = FFI()
 
@@ -18,6 +19,8 @@ char *get_tcl_version();
 #define TCL_EVAL_DIRECT ...
 #define TCL_EVAL_GLOBAL ...
 
+#define TCL_DONT_WAIT ...
+
 typedef unsigned short Tcl_UniChar;
 typedef ... Tcl_Interp;
 typedef ...* Tcl_ThreadId;
@@ -102,6 +105,15 @@ int Tcl_DoOneEvent(int flags);
 int Tk_GetNumMainWindows();
 """)
 
+if sys.platform.startswith("openbsd"):
+    incdirs = ['${MODTCL_INCDIR}', '${MODTK_INCDIR}', '${X11BASE}/include']
+    linklibs = ['tk85', 'tcl85']
+    libdirs = ['${LOCALBASE}/lib', '${X11BASE}/lib']
+else:
+    incdirs=['/usr/include/tcl'],
+    linklibs=['tcl', 'tk'],
+    libdirs = []
+
 tklib = tkffi.verify("""
 #include <tcl.h>
 #include <tk.h>
@@ -109,6 +121,7 @@ tklib = tkffi.verify("""
 char *get_tk_version() { return TK_VERSION; }
 char *get_tcl_version() { return TCL_VERSION; }
 """,
-include_dirs=['/usr/include/tcl'],
-libraries=['tcl', 'tk'],
+include_dirs=incdirs,
+libraries=linklibs,
+library_dirs = libdirs
 )
